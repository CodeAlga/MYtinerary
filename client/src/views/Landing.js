import React from "react";
//import Footer from "../components/Footer";
import Home from "../components/Home";
import Carousel from "../components/Carousel";
import Header from "../components/Header";

export default function Landing(props) {
  return (
    <div className="landingBox">
      <div className="landingForeground">
        <Header />
        <Home />
        <Carousel {...props} />
        {/* <Footer alignSelf="flex-end" /> */}
      </div>
      <div className="landingBackground"></div>
    </div>
  );
}
