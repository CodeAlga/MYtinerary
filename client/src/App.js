import React, { Component } from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import Landing from "./views/Landing";
import Cities from "./views/Cities";
import RegisterView from "./views/RegisterView";
import LoginView from "./views/LoginView";
import CityView from "./views/CityView";

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      listCities: [],
      listId: []
    };
  }

  componentDidMount() {
    fetch("cities/all")
      .then((res) => res.json())
      .then((data) => {
        this.setState({
          listCities: data
        });
        this.getId(this.state.listCities);
      });
  }

  getId = (arr) => {
    let arrayId = [];
    arr.map((item) => {
      arrayId.push(item._id);
      return arrayId;
    });
    this.setState({ listId: arrayId });
  };

  render() {
    const classes = (theme) => ({
      root: {
        height: "100vh"
      }
    });

    return (
      <BrowserRouter>
        <div className="App">
          <div className={classes.root}>
            <Switch>
              <Route
                exact
                path="/"
                render={(props) => (
                  <Landing
                    listCities={this.state.listCities}
                    {...this.state.props}
                  />
                )}
              />
              <Route
                path="/cities"
                render={(props) => (
                  <Cities
                    listCities={this.state.listCities}
                    {...this.state.props}
                  />
                )}
              />
              <Route path="/register" component={RegisterView} />
              <Route path="/login" component={LoginView} />
              <Route path="/city/:listCities._id" component={CityView} />
            </Switch>
          </div>
        </div>
      </BrowserRouter>
    );
  }
}

export default App;
