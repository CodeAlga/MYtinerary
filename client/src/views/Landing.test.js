/* import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import App from "../App";
import Landing from "./Landing"; */

//import { shallow } from "enzyme";

//import toJson from "enzyme-to-json";

/* test("renders learn react link", () => {
  const { getByText } = render(<App />);
});
describe("what is the unit under test?", () => {
  it("Landing component should render without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Landing />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
}); */

// it("Landing component should render without crashing", () => {
//   shallow(<Landing />);
// });

// it("landing component snapshot with Enzyme", () => {
//   const wrapper = shallow(<Landing />);
//   expect(toJson(wrapper)).toMatchSnapshot();
// });
