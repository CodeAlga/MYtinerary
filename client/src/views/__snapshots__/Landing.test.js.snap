// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing component snapshot with Enzyme 1`] = `
<div
  className="landingBox"
>
  <div
    className="landingForeground"
  >
    <Header />
    <Home />
    <WithStyles(SimpleSlider) />
  </div>
  <div
    className="landingBackground"
  />
</div>
`;
